pkg/server/testing/fake_cni_plugin.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testing
+
+import (
+	"sync"
+
+	"github.com/kubernetes-incubator/cri-o/pkg/ocicni"
+)
+
+// FakeCNIPlugin is a fake plugin used for test.
+type FakeCNIPlugin struct {
```

Comment:
Take a look at https://github.com/xlgao-zju/cri-containerd/blob/d9a3854d2595e98604a3ef6a46de11b9a68d992c/pkg/server/testing/fake_execution_client.go, we need to do similar thing.

All functions should track down call details.
All functions should be able to inject error.
We can use some test mock generator, but it's simple enough to write ourselves.